src/pages/Form/SingleForm.tsx

```diff
@@ -14,6 +14,7 @@ const SingleForm = () => {
   const { data, isFetching } = useGetFormQuery({ id, jwt });
   const name = data?.name;
   const description = data?.description || "";
+  const type = data?.type || "";
   const questions = data?.questionIds || [];
```

Comment:
Here you're setting name, description, ... to "" (empty string) when data is not available (an falsy value).
But the backend never returns that as if there is no for a 404 error is thrown instead of null.

So, I'd suggest that we just check if that is available (to prevent typescript from screaming at you) or if `isFeching` and render the loading spinner otherwise the data should be available.

````js

if (data! || isFetching) {
    return (
      <div className="h-[50vh] flex items-center justify-center">
        <Loader />
      </div>
    );
  }
````

I also think that name, description,... variables are kinda a lot may be you can consider destructuring instead.